tests/KdybyTests/CsobPaymentGateway/HttpClientMock.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+namespace KdybyTests\CsobPaymentGateway;
+
+use GuzzleHttp\Psr7\Response;
+use Kdyby\CsobPaymentGateway\Http\GuzzleClient;
+use Kdyby\CsobPaymentGateway\IHttpClient;
+use Psr\Http\Message\ResponseInterface;
+
+
+
+/**
+ * @author Jiří Pudil <me@jiripudil.cz>
+ */
+class HttpClientMock implements IHttpClient
+{
+
+	/**
+	 * @var GuzzleClient
+	 */
+	private $guzzle;
+
+
+
+	public function __construct()
+	{
+		$this->guzzle = new GuzzleClient();
+	}
+
+
+
+	/**
+	 * @param string $method
+	 * @param string $url
+	 * @param array $headers
+	 * @param string $body
+	 * @return ResponseInterface
+	 */
+	public function request($method, $url, $headers, $body)
+	{
+		$parsedUrl = parse_url($url);
+		$path = isset($parsedUrl['path']) ? $parsedUrl['path'] : '';
+
+		if ($path === '/api/v1.5/payment/init' && $method === 'POST') {
```

Comment:
tady by to možná šlo maličko unvierzálněji, ale to můžem řešit až se budou psát testy na ty další rq